lib/Net/Stripe.pm

```diff
@@ -1545,10 +1549,70 @@ method _post(Str $path, $obj?) {
     return $self->_make_request($req);
 }
 
+sub BUILD {
+    my ( $self, $args ) = @_;
+
+    my $message = "The API version %s (%s) is not supported by this version of Net::Stripe. " .
+    "This version only supports %s or before. " .
+    "Please download a version-appropriate branch at https://github.com/lukec/stripe-perl/branches.";
+
+    if (defined($self->api_version)) {
+        if ($self->api_version gt $_MAX_API_VERSION) {
+            die sprintf($message,
+                "requested",
+                $self->api_version,
+                $_MAX_API_VERSION,
+            );
+        }
+    } else {
+        my $account_api_version = $self->_get_account_api_version();
+        if ($account_api_version gt $_MAX_API_VERSION) {
+            die sprintf($message,
+                "for your Stripe account",
+                $account_api_version,
+                $_MAX_API_VERSION,
+            );
+        }
+    }
+}
+
+# i really wish that the Stripe API had a ping-like method so that we
+# do not have to perform an extraneous request simply to check the
+# Stripe-Version header on the response.
+# for now, we will retrieve the balance, without specifying the API
```

Comment:
Looks ok to me since it's in the minimal set of permissions https://support.stripe.com/questions/security-permissions-and-access-levels-when-connecting-your-stripe-account-to-a-third-party-platform